### PR TITLE
QRS-10, QRS-15, Add History page and simple unit tests

### PR DESCRIPTION
**Pull request by VitalInnovator**

### CODE REVIEW

**Comment by roszcz** [comment]
@VitalInnovator thanks! I'll test that today. 

Would it be possible to add a test that checks for issues like what happened in: https://github.com/Nospoko/data-quality-platform/pull/9? This is mostly to prevent regressions.

**Comment by roszcz** [comment]
@VitalInnovator can you also add some documentation about how to run the tests/linters locally?

**Comment by VitalInnovator** [comment]
> @VitalInnovator thanks! I'll test that today.
> 
> Would it be possible to add a test that checks for issues like what happened in: #9? This is mostly to prevent regressions.

Yes, I'll add

**Comment by VitalInnovator** [comment]
> @VitalInnovator can you also add some documentation about how to run the tests/linters locally?

It's a good question how run it with Docker))

**Comment by roszcz** [comment]
@VitalInnovator If you need the DB running, the process could look something like this:

```sh
docker compose up
docker compose run api yarn test
```

You can change `yarn test` to something else - ideally the same set of commands that's running in the github workflow.

**Comment by roszcz** [comment]
I found a couple of issues, I captured all of them in a video:

https://github.com/Nospoko/data-quality-platform/assets/8056825/52dfaf88-38b4-4adb-82ef-58d971098a88

Here's a list:

1. I think the history list does not contain the last processed record. This is evident for the first one you change - history is still empty after doing that.
2. (end of the video) When I change the selection in the zoom view on the history page, it gets updated only in the small view in the background, but not on the zoomed view.

File: next/src/modules/homeChart/components/ZoomView.tsx [hunk 1]
```diff
@@ -32,12 +39,54 @@ const ZoomView: React.FC<Props> = ({
...
   return (
     <>
+      <Modal
+        title="Confirmation"
+        centered
+        visible={isConfirmModal}
+        onOk={handleConfirm}
+        onCancel={handleCancel}
+      >
+        <p>Are you sure you want to change the feedback?</p>
```
Comment (roszcz) [inline next/src/modules/homeChart/components/ZoomView.tsx]:
```suggestion
        <p>Are you sure you want to change the label?</p>
```

This question makes more sense that way

**Comment by roszcz** [comment]
Hi @VitalInnovator, I'm not sure what changed, but the history page stopped working for me:

https://github.com/Nospoko/data-quality-platform/assets/8056825/d33d8ce4-23a1-46d9-9b1c-ff35edd46a39

The error says to be about this: https://nextjs.org/docs/messages/react-hydration-error. Please let me know when you're online if this is something you can fix or if it's something wrong with my environment 🤔

**Comment by VitalInnovator** [comment]
> Hi @VitalInnovator, I'm not sure what changed, but the history page stopped working for me:
> 
>  Screen.Recording.2023-07-21.at.11.36.33.mov 
> The error says to be about this: https://nextjs.org/docs/messages/react-hydration-error. Please let me know when you're online if this is something you can fix or if it's something wrong with my environment 🤔

This is an issue with css-in-js. It doesn't have any apparent consequences. I'll work on fixing it and recheck the configurations.

**Comment by VitalInnovator** [comment]
> I found a couple of issues, I captured all of them in a video:
> 
>  Screen.Recording.2023-07-20.at.20.24.33.mov 
> Here's a list:
> 
> 1. I think the history list does not contain the last processed record. This is evident for the first one you change - history is still empty after doing that.
> 2. (end of the video) When I change the selection in the zoom view on the history page, it gets updated only in the small view in the background, but not on the zoomed view.

@roszcz Fixed

**Comment by VitalInnovator** [comment]
@roszcz I slightly changed the fetch logic for the history page. Could you please check if this page works for you now?

**Comment by roszcz** [comment]
> This is an issue with css-in-js. It doesn't have any apparent consequences. I'll work on fixing it and recheck the configurations.

The main problem the video shows is that there is nothing showing up on the history page. I don't know if that's the consequence of the error - but even if it's completely unrelated, I think we shouldn't allow unaccounted errors in the console. I just tried with the latest changes and it's the same 🤔

**Comment by roszcz** [comment]
@VitalInnovator Looks like the history tab is working again for me, but I don't really now if it's something I did that fixed it. I was trying to restart/refresh everything and suddenly the records are back - let me know if you have any idea about what's that all about 😅 

If the issue doesn't come back for me, let's assume that it was specific to my environment and ignore it 👍

**Review by roszcz — APPROVED** [review]

**Pull request merged**